datarobot_batch_scoring/reader.py

```diff
@@ -379,6 +379,8 @@ def investigate_encoding_and_dialect(dataset, sep, ui, fast=False,
     else:
         opener = open
     with opener(dataset, 'rb') as dfile:
+        line_to_sniff = dfile.readline()
```

Comment:
yeah, you're right. I double checked in pandas source and they sniff the first line only for the delimiter.

I think, given how unreliable the sniffer is for edge cases like this, we need to allow users to specify the dialect options -- delimiter, qoutechar, lineterminator, etc. We can try and sniff the csv and provide a user-friendly output of what we sniffed with a message, like "We tried to automatically detect this and this is what we've got. If this is wrong, use `--delimiter` (or other options)"

For example, `pandas.read_csv` allows you to override every dialect property. They sniff the csv only to detect the delimiter, everything else is passed as an argument with a sane default.